File: First Unique Character in a String/First_Unique_Characte_ in_a_String.java
class Solution {
    public int firstUniqChar(String s) {
        Map<Character, Integer> mpp = new HashMap<>();
        for (char c : s.toCharArray()) {
            mpp.put(c, mpp.getOrDefault(c, 0) + 1);
        }
        int n = s.length();
        for (int i = 0; i < n; i++) {
            if (mpp.get(s.charAt(i)) == 1) {
                return i;
            }
        }
        return -1;
    }
}

File: First Unique Character in a String/First_Unique_Characte_ in_a_String.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map<char,int> mpp;
        for(auto i: s) {
            mpp[i]++;
        }
        int n = s.size();
        for(int i = 0; i<n; i++) {
            if(mpp[s[i]] == 1) return i;
        }
        return -1;
    }
};